### PR TITLE
UPDATED THE UI: Live price for the homepage chart

### LINKED ISSUE #49: UI: Live price for the homepage chart
## ⚠️ **Reading contributors guidelines to get assigned is MANDATORY!**
[Read contributor guidelines](https://github.com/walnuthq/op-scan/issues/1)

## Task description
In the chart on the homepage, the "Price" always says $2 because it is hardcoded. Fetch the live data from Coinbase API and display the accurate price for each date in the chart.

CleanShot 2024-07-29 at 16 25 38@2x

### PR DESCRIPTION
**Pull request by Dprof-in-tech**
This pr resolves #49 

- [x] please use standard fetch, do not add axios or another library on top.
- [x] fetch the OP token price, later in the future we will need to add this as a config option but it's out of scope at the moment.
- [x] please stay within the scope of this issue which is very limited, in particular do not add loading indicators, auto refresh or additional controls to display another range as this is not needed at the moment.
- [x] use the shadcn/ui charts component
- [x] made sure there are no build errors or anything

Media files to show the result of this pr

image

### CODE REVIEW

File: src/components/pages/home/transaction-history.tsx [hunk 1]
```diff
@@ -1,71 +1,122 @@
...
+import { ChartContainer, type ChartConfig } from "@/components/ui/chart";
 import { formatPrice } from "@/lib/utils";
 import { subDays, format, parseISO } from "date-fns";
 
-const data = Array.from({ length: 15 }, (_, i) => i).map((i) => {
-  const date = subDays(new Date(), 15 - i);
-  return {
-    name: format(date, "MMM d"),
-    date: format(date, "yyyy-MM-dd"),
-    transactionsCount: Math.floor(Math.random() * 200) + 400,
```
Comment (saimeunt) [inline src/components/pages/home/transaction-history.tsx:20]: This placeholder datapoint is still needed, please reintroduce it.

File: src/components/pages/home/transaction-history.tsx [hunk 2]
```diff
@@ -1,71 +1,122 @@
...
+    label: "Mobile",
+    color: "#60a5fa",
+  },
+} satisfies ChartConfig;
 
-const CustomTooltip = ({ active, payload }: TooltipProps<number, string>) => {
+type DataPoint = {
+  name: string;
+  date: string;
+  price: number;
```
Comment (saimeunt) [inline src/components/pages/home/transaction-history.tsx:31]: Missing the `transactionsCount` here.

File: src/components/pages/home/transaction-history.tsx [hunk 3]
```diff
@@ -1,71 +1,122 @@
...
+};
+
+const CustomTooltip = ({ active, payload }: any) => {
```
Comment (saimeunt) [inline src/components/pages/home/transaction-history.tsx]: Why removing typing here? We don't want any, please reintroduce `TooltipProps<number, string>`.

File: src/components/pages/home/transaction-history.tsx [hunk 4]
```diff
@@ -1,71 +1,122 @@
...
   if (active && payload && payload.length) {
     return (
       <div className="rounded-md border bg-popover px-3 py-1.5 text-sm text-popover-foreground shadow-md">
-        <p className="test-xs">
+        <p className="text-xs">
           {format(parseISO(payload[0].payload.date), "EEEE, MMMM d, yyyy")}
         </p>
-        <p>Transactions: {payload[0].value}k</p>
```
Comment (saimeunt) [inline src/components/pages/home/transaction-history.tsx:32]: We do want the transactions count displayed in the tooltip.

File: src/components/pages/home/transaction-history.tsx [hunk 5]
```diff
@@ -1,71 +1,122 @@
...
-          />
-        </LineChart>
-      </ResponsiveContainer>
-    </CardContent>
-  </Card>
-);
+const TransactionHistory = () => {
+  const [data, setData] = useState<DataPoint[]>([]);
+
+  useEffect(() => {
```
Comment (saimeunt) [inline src/components/pages/home/transaction-history.tsx]: Please avoid client-side data fetching as it's unnecessary here and can be refactored in using React Server Components.
Build the `DataPoint` array in the top level homepage component in `components/pages/home/index.tsx` and pass it as a prop in the `TransactionHistory` component.

**Review by saimeunt — CHANGES_REQUESTED** [review]
@Dprof-in-tech this is heading in the right direction, nice work, please fix the small problems pointed out (reintroduce transactionsCount and use server-side data fetching) and we can merge this!

File: src/components/pages/home/transaction-history.tsx (hunk 3 above)
Comment (Dprof-in-tech) [reply src/components/pages/home/transaction-history.tsx]: fixed

File: src/components/pages/home/transaction-history.tsx (hunk 2 above)
Comment (Dprof-in-tech) [reply src/components/pages/home/transaction-history.tsx:31]: reintroduced chief

**Comment by Dprof-in-tech** [comment]
i've fixed the code as required in your review. here's a media file to show

would love to be assigned to another once this one is merged if possible

thanks chief

image

File: src/components/pages/home/transaction-history.tsx [hunk 6]
```diff
@@ -28,54 +29,27 @@ type DataPoint = {
...
-        }),
-      );
-
-      setData(responses);
-    };
-
-    fetchData();
-  }, []);
+const TransactionHistory = ({ history }: { history: DataPoint[] }) => {
+  const [data, setData] = useState<DataPoint[]>(history);
```
Comment (saimeunt) [inline src/components/pages/home/transaction-history.tsx]: I think this is no longer necessary and you can use the history prop directly.

**Review by saimeunt — CHANGES_REQUESTED** [review]
@Dprof-in-tech nice work on the requested changes, one final change requested and we can merge.

File: src/components/pages/home/transaction-history.tsx (hunk 6 above)
Comment (Dprof-in-tech) [reply src/components/pages/home/transaction-history.tsx]: done chief

**Review by saimeunt — CHANGES_REQUESTED** [review]
@Dprof-in-tech You made the requested changes but your merge commit is wrong, we moved `fetchHomePageData` to a new function named `fetchHomeData` in `components/pages/home/fetch-home-data.ts`.

You must use this function in `components/pages/home/index.tsx` as `fetchHomePageData` has been removed from the codebase.
What you need to do instead is move your `fetchTransactionHistory` function to `components/pages/home/fetch-home-data.ts` and use it inside `fetchHomeData`.

Your commit should not introduce any changes to `lib/fetch-data.ts` as you currently have introduced regressions by using your outdated version of this file instead of the main branch version.

**Comment by Dprof-in-tech** [comment]
Okay

Could you give me a few minutes to effect this change real quick so you
could take a look??

**Comment by Dprof-in-tech** [comment]
Hey chief @saimeunt 
I just fixed the issue

Please review

Thank you

**Review by saimeunt — APPROVED** [review]

**Pull request merged**